src/services/lease-service/tests/index.test.ts

```diff
@@ -248,6 +311,34 @@ describe('lease-service', () => {
     })
   })
 
+  describe('GET /listing/:listingId/applicants/details', () => {
+    it('responds with 404 if no listing found', async () => {
+      const getListingSpy = jest
+        .spyOn(listingAdapter, 'getListingById')
+        .mockResolvedValue(undefined)
```

Comment:
```suggestion
        .mockResolvedValueOnce(undefined)
```
Isolerar och tar bort risken att state för spies ska läcka in i andra tester
skulle säga det gäller för alla spies vars returvärden specas inuti testerna